### PR TITLE
Add Dsa unit test functions.

### PR DESCRIPTION
**Pull request by jrblixt**
Configure options: --enable-dsa --enable-keygen.

### CODE REVIEW

File: tests/api.c [hunk 1]
```diff
@@ -122,6 +122,22 @@
     #include <wolfssl/wolfcrypt/hc128.h>
 #endif
 
+#ifndef NO_DSA
+    #include <wolfssl/wolfcrypt/dsa.h>
+    #ifndef BYTE1024
+        #define BYTE1024 1024
```
Comment (cconlon) [inline tests/api.c]: For consistency in naming, I may call these ONEK_BUF and TWOK_BUF (to match the FOURK_BUF name).

File: tests/api.c [hunk 2]
```diff
@@ -8471,6 +8487,379 @@ static int test_wc_Hc128_Process (void)
...
+        }
+    } /* END Private Key  */
+
+    printf(resultFmt, priv == 0 ? passed : failed);
+
+    printf(testingFmt, "wc_DsaPublicKeyDecode()");
+    if (ret == 0) {
+        idx = 0; /* Reset */
+        pub = wc_DsaPublicKeyDecode(tmp, &idx, &key, bytes);
+    /* Test bad args. */
```
Comment (cconlon) [inline tests/api.c]: This line needs an extra tab.

File: tests/api.c [hunk 3]
```diff
@@ -8471,6 +8487,379 @@ static int test_wc_Hc128_Process (void)
...
+
+    wc_FreeDsaKey(&genKey);
+#endif
+    return ret;
+} /* END test_wc_MakeDsaKey */
+
+/*
+ * Testing wc_DsaKeyToDer()
+ */
+static int test_wc_DsaKeyToDer (void)
```
Comment (cconlon) [inline tests/api.c]: Should we test a known "vector" here, where we read in a DSA key from file into a DsaKey struct, then call DsaKeyToDer() and compare to the expected byte array?

Comment (jrblixt) [inline tests/api.c]: This should show outdated as well. I've corrected it the issue. Thank you.

Comment (jrblixt) [inline tests/api.c]: This issue has been corrected and should show outdated as well. Thank you.

File: tests/api.c [hunk 4]
```diff
@@ -8471,6 +8487,390 @@ static int test_wc_Hc128_Process (void)
...
+} /* END test_wc_InitDsaKey */
+
+/*
+ * Testing wc_DsaSign() and wc_DsaVerify()
+ */
+static int test_wc_DsaSignVerify (void)
+{
+    int     ret = 0;
+
+#if !defined(NO_DSA) && (defined(USE_CERT_BUFFERS_1024)\
```
Comment (cconlon) [inline tests/api.c]: Is there a reason we don't test this with files if USE_CERT_BUFFERS_1024 or USE_CERT_BUFFERS_2048 are not defined?  We could use ./certs/dsa2048.der (which matches dsa_key_der_2048[]).

Same applies to other functions in this PR.

File: tests/api.c (hunk 3 above)
Comment (cconlon) [inline tests/api.c]: Thanks

**Comment by jrblixt** [comment]
Retest this please Jenkins.

**Pull request merged**